contracts/LockProxySwitcheo.scilla

```diff
@@ -0,0 +1,605 @@
+end
+
+procedure validateAddrLength(proxyAddr: ByStr)
+  length = builtin strlen proxyAddr;
+  is_addr = builtin eq length uint32_20;
+  match is_addr with
+  | True =>
+  | False =>
+    e = IllegalRegisterAssetArgs;
+    ThrowError e
+  end
+end
+
+(* can be used both for validating targetProxyHash and toAddress and fromAssetHash *)
+procedure ValidataHashLength(hash: ByStr)
+  hash_length = builtin strlen hash;
+  zero_length = builtin eq hash_length uint32_zero;
+  match zero_length with
+  | True =>
+    e = EmptyHashStr;
+    ThrowError e
+  | False =>
+  end
+end
+
+(* Mark an asset as registered by associating it to a specified Switcheo TradeHub proxy and asset hash*)
+procedure MarkAssetAsRegistered(tokenAddr: ByStr20, proxyAddr: ByStr, toAssetHash: ByStr)
+  validateAddrLength proxyAddr;
+  exist <- exists register[tokenAddr];
+  match exist with
+  | True =>
+    e = AssetAlreadyRegistered;
+    ThrowError e
+  | False =>
+    x = serialize_proxy_addr_to_asset_hash proxyAddr toAssetHash;
+    value = builtin keccak256hash x;
+    register[tokenAddr] := value
+  end
+end
+
+(* Validates that an asset registration matches the given params *)
+(* @param tokenAddr: the address of the asset to check *)
+(* @param proxyAddr: the expected proxy address on Swticheo TradeHub *)
+(* @param toAssetHash: the expected asset hash on Swticheo TradeHub *)
+procedure ValidateAssetRegistration(tokenAddr: ByStr20, proxyAddr: ByStr, toAssetHash: ByStr)
+  validateAddrLength proxyAddr;
+  x = serialize_proxy_addr_to_asset_hash proxyAddr toAssetHash;
+  value = builtin keccak256hash x;
+  value_stored_o <- register[tokenAddr];
+  match value_stored_o with
+  | Some value_stored =>
+    registered = builtin eq value_stored value;
+    match registered with
+    | True =>
+    | False =>
+      e = AssetNotRegistered;
+      ThrowError e
+    end
+  | None => 
+    e = AssetNotRegistered;
+    ThrowError e
+  end
+end
+
+(* Validates the asset registertion and calls the CCM contract *)
+procedure lockProcedure(
+  tokenAddr: ByStr20,
+  targetProxyHash: ByStr, 
+  toAssetHash: ByStr, 
+  fromAddr: ByStr20, 
+  toAddress: ByStr, 
+  amount: Uint256, 
+  feeAmount: Uint256, 
+  feeAddr: ByStr
+)
+  validateAddrLength targetProxyHash;
+  ValidataHashLength toAssetHash;
+  ValidataHashLength toAddress;
+  amount_uint128_o = builtin to_uint128 amount;
+  match amount_uint128_o with
+  | Some amount_uint128 =>
+    validateAmount amount_uint128;
+    fee_amount_uint128_o = builtin to_uint128 feeAmount;
+    match fee_amount_uint128_o with
+    | Some fee_amount_uint128 =>
+      validateFeeAmount fee_amount_uint128 amount_uint128;
+      ValidateAssetRegistration tokenAddr targetProxyHash toAssetHash;
+      from_asset_hash = builtin to_bystr tokenAddr;
+      from_address = builtin to_bystr fromAddr;
+      increaseNonce;
+      n <- nonce;
+      transfer_tx_args = TransferTxArgs from_asset_hash toAssetHash toAddress from_address amount feeAmount feeAddr n;
```

Comment:
let's standardize the coding convention. Sometimes is xx_xxx_xx, something is xxXxxXxx

Reply:
done!